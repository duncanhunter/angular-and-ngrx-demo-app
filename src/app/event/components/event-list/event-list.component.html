<ul>
  <li *ngFor="let attendee of attendees">{{attendee.name}}</li>
</ul>
