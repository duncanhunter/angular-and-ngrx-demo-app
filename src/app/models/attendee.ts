export interface Attendee {
  id?: number;
  name: string;
  guests: number;
  attending: boolean;
}
