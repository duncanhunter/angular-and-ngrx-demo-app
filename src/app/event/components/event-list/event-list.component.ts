import { Component, Input } from '@angular/core';
import { Attendee } from '../../../models';

@Component({
  selector: 'app-event-list',
  templateUrl: './event-list.component.html',
  styleUrls: ['./event-list.component.scss']
})
export class EventListComponent {
  @Input() attendees: Attendee[];
}
