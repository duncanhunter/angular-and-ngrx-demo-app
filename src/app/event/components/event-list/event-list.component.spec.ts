import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Attendee } from './../../../models';

import { EventListComponent } from './event-list.component';

describe('EventListComponent', () => {
  let component: EventListComponent;
  let fixture: ComponentFixture<EventListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ EventListComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(EventListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should have no attendee on load', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have 1 attendee on load', () => {
    component.attendees = [
      {name: 'Duncan', attending: true, guests: 2}
    ] as Attendee[];
    fixture.detectChanges();

    expect(component).toMatchSnapshot();
  });
});
