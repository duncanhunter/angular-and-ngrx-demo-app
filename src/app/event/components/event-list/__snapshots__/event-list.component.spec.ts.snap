// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListComponent should have 1 attendee on load 1`] = `
EventListComponent {
  "attendees": Array [
    Object {
      "attending": true,
      "guests": 2,
      "name": "Duncan",
    },
  ],
}
`;

exports[`EventListComponent should have no attendeea on load 1`] = `EventListComponent {}`;
